api/x509cert_test.go

```diff
@@ -257,26 +269,49 @@ func TestPostX509Certificate(t *testing.T) {
 			expectedCert: nil,
 			CSR:          testGoodcsrRsa,
 		},
+		"requestTimeout": {
+			ctx:          timeoutCtx,
+			KeyUsages:    combineKeyUsage,
+			maxValidity:  map[string]uint64{config.X509CertEndpoint: 3600},
+			validity:     3600,
+			KeyMeta:      &proto.KeyMeta{Identifier: "x509id1"},
+			expectedCert: nil,
+			CSR:          testGoodcsrRsa,
+		},
+		"requestCanceled": {
+			ctx:          cancelCtx,
+			KeyUsages:    combineKeyUsage,
+			maxValidity:  map[string]uint64{config.X509CertEndpoint: 3600},
+			validity:     3600,
+			KeyMeta:      &proto.KeyMeta{Identifier: "x509id1"},
+			expectedCert: nil,
+			CSR:          testGoodcsrRsa,
+		},
 	}
 	for label, tt := range testcases {
 		tt := tt
 		label := label
+		if tt.ctx == nil {
+			tt.ctx = ctx
+		}
+		if label == "requestTimeout" {
+			time.Sleep(timeout)
```

Comment:
can we instead have a sleep in the server mock?

Reply:
The only concern I have with that is it will sleep for each request causing the test to run for ~10-12 seconds.

Reply:
can we pass the timeout value while initializing the mock server?